java/.ci/Jenkinsfile

```diff
@@ -803,6 +814,13 @@ pipeline {
     				notifyBackToNormal: 'true'
                 }
             }
+            catchError {
+                sh '''
+                if [ $GIT_BRANCH_LOCAL != "main" ];then
+                    gsutil rm -r $GCS_STAGING_LOCATION 2> /dev/null || true
+                fi
```

Comment:
Would the catchError block only run when the pipeline fails? Ideally we want to remove the GCS location in successful runs as well to avoid cluterring the GCS bucket.

Reply:
No, this block will run every time. Please refer - https://dataproctemplatesci.com/job/dataproc-templates-build/job/ad-hoc/job/integration-tests-java/313/console